main.go

```diff
@@ -191,6 +193,21 @@ func main() {
 		log.Fatal(err)
 	}
 
+	// Creates the output upload exclusions file, required by the JobCompose InitFromJob method.
+	excludeFile, err := os.Create(dcompose.UploadExcludesFilename)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer excludeFile.Close()
+
+	excludeFileContents := strings.Join(job.ExcludeArguments(), "\n")
+
+	_, err = io.Copy(excludeFile, strings.NewReader(excludeFileContents))
+	if err != nil {
+		log.Fatal(err)
+	}
+	excludeFile.Close()
```

Comment:
It's weird that you have this as well as the `defer` one above. Maybe better to split this file creation all out into a dedicated function and keep only the `defer` one?

Reply:
Of course I guess we do the same thing for the docker-compose file below, so maybe it's not worth the bother? dunno

Reply:
I don't know either (and yes: this was clip-board inheritance from the docker-compose code below).